src/swarm/neo/client/ClientSocket.d

```diff
@@ -94,11 +96,12 @@ public class ClientSocket
         if (this.socket_.fd >= 0)
         {
             /*
-             * TODO: Log a warning if this happens.
              * The socket should not exist or there is a bug. However, this
              * situation is too critical to assert or ignore it and leak file
              * descriptors.
              */
+            Log.lookup(this.classinfo.name).warn("connect: The socket " ~
```

Comment:
Huh, that's a much nicer approach than the typical module logger. I wonder why we don't do this more commonly?